### ISSUE TITLE
[QUESTION] Virtual Machine Snapshots via API

### ISSUE THREAD

**Issue by neotheonce** [harvester/harvester#7050]
Hello,

I am trying to manage virtual machine snapshots via Harvester API 1.3. I  am referring following document
https://docs.harvesterhci.io/v1.3/api/delete-namespaced-virtual-machine-backup/ .  Wrote small python program for API calls. I am getting 500 error when I am trying to delete the snapshot

```
#!/usr/bin/env python3
import requests
import json 
...
...
...
url= https://harvester.xxxxxx.net/apis/harvesterhci.io/v1beta1/namespaces/api/virtualmachinebackups/ansible-26-11-2024-1
headers = {'Content-type': 'application/json'}
postdata = {"apiVersion":"harvesterhci.io/v1beta1", "kind":"VirtualMachineBackup"}
harv_session.delete(url, headers=headers, data=json.dumps(postdata))
```

Output :
{'kind': 'Status', 'apiVersion': 'v1', 'metadata': {}, 'status': 'Failure', 'message': 'no kind "VirtualMachineBackup" is registered for version "harvesterhci.io/v1beta1" in scheme "k8s.io/apimachinery/pkg/apis/meta/internalversion/scheme/register.go:27"', 'code': 500}

Am I sending correct JSON body? I tried to search in API docs but could not  find any references. 

Any help is appreciated

--
Neo

**Comment by ihcsim** [comment]
Can you confirm the server-side version of the `VirtualMachineBackup` CRD? Try running:

```sh
$ kubectl get crd virtualmachinebackups.harvesterhci.io -ojsonpath='{.spec.versions[*].name}'
```

**Comment by neotheonce** [comment]
Shows v1beta1 
 \# kubectl get crd virtualmachinebackups.harvesterhci.io -ojsonpath='{.spec.versions[*].name}'
v1beta1
#

**Comment by brandboat** [comment]
Hi @neotheonce, I just tested the virtualmachinebackups API through curl with harvester v1.3.2, and it works fine. For further investigation, could you please provide [support-bundle](https://docs.harvesterhci.io/v1.3/troubleshooting/harvester#generate-a-support-bundle) ? Thanks.
```
curl 'https://<harvester_ip>/apis/harvesterhci.io/v1beta1/namespaces/default/virtualmachinebackups/mysnapshot' \
   -X 'DELETE' \
   -H 'Authorization: Bearer <my-bearer-token>' \
   --insecure
```

**Comment by neotheonce** [comment]
Yep I worked when I did not send JSON body  {"apiVersion":"harvesterhci.io/v1beta1", "kind":"VirtualMachineBackup"}
Not sure why it's mentioned required in documentation. 
https://docs.harvesterhci.io/v1.3/api/delete-namespaced-virtual-machine-backup


--
Thanks,
Neo

**Comment by ihcsim** [comment]
@neotheonce Glad you figure it out. The request JSON body is meant to describe the GVK of the `v1/DeleteOption` subresource, not the `VirtualMachineBackup` resource. E.g., one would use it to change the delete propagation policy. The request path parameters are the important ones which define the GVR of the `VirtualMachineBackup` resource.

If you have thoughts on how the docs can be improved, please raise an issue in the https://github.com/harvester/docs repo, and feel free to close this one. Thanks for supporting Harvester!

**Comment by neotheonce** [comment]
Closing

--
Neo
